01_Data/src/layers/sequelize/model/TransactionEvent/Transaction.ts

```diff
@@ -38,9 +40,15 @@ export class Transaction extends Model implements TransactionType {
   @Column(DataType.BOOLEAN)
   declare isActive: boolean;
 
-  @HasMany(() => TransactionEvent)
+  @HasMany(() => TransactionEvent,
+    { as: Transaction.TRANSACTION_EVENTS_ALIAS, foreignKey: 'transactionDatabaseId' })
   declare transactionEvents?: TransactionEventRequest[];
 
+  // required only for filtering, should not be used to pull transaction events
+  @HasMany(() => TransactionEvent,
+    { as: Transaction.TRANSACTION_EVENTS_FILTER_ALIAS, foreignKey: 'transactionDatabaseId' })
+  declare transactionEventsFilter?: TransactionEventRequest[];
```

Comment:
Could you elaborate on why this field is needed? I've looked over both PRs and I have yet to understand fully...

Reply:
not sure why it doent show in the diff, but the EVENTS_FILTER_ALIAS is being used in the `TransactionQueryBuilder`:

```
buildQuery(params: QueryParams, endedOnly?: boolean): any {
    const modelsToInclude: any[] = [
      this.createMspFilters(params.mspCountryCode, params.mspPartyId),
      this.createCpoFilters(params.cpoCountryCode, params.cpoPartyId),
      MeterValue,
      Evse,
      ...(endedOnly
        ? [
            {
              model: TransactionEvent,
              as: Transaction.TRANSACTION_EVENTS_FILTER_ALIAS,
              attributes: ['eventType'],
              where: {
                eventType: TransactionEventEnumType.Ended,
              },
            },
          ]
        : []),
    ];
```

Reply:
why do we need a separate alias though, cant we just use the same one @danielyoo227 ?